dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -3033,20 +3029,18 @@ Spdp::SpdpTransport::handle_input(ACE_HANDLE h)
 
         ParameterList plist;
         if (data.smHeader.flags & (FLAG_D | FLAG_K_IN_DATA)) {
-          ser.swap_bytes(!ACE_CDR_BYTE_ORDER); // read "encap" itself in LE
-          CORBA::UShort encap, options;
-          if (!(ser >> encap) || (encap != encap_LE && encap != encap_BE)) {
+          DCPS::EncapsulationHeader encap;
+          if (!(ser >> encap) || (encap.kind() != DCPS::EncapsulationHeader::KIND_PL_CDR_LE &&
+                                  encap.kind() != DCPS::EncapsulationHeader::KIND_PL_CDR_BE)) {
             if (DCPS::DCPS_debug_level > 0) {
               ACE_ERROR((LM_ERROR,
                         ACE_TEXT("(%P|%t) ERROR: Spdp::SpdpTransport::handle_input() - ")
                         ACE_TEXT("failed to deserialize encapsulation header for SPDP\n")));
             }
             return 0;
           }
```

Comment:
This just needs to do the equivalent of what `swap_bytes` was doing before. This would be turning `encap` into a `Encoding` and then set use that to set `ser`. Unfortunately just deserializing doesn't do all that. I think at some point I wanted it to that when I created `EncapsulationHeader`, but this allows for more possible interaction with the encap object if needed.

```suggestion
          DCPS::EncapsulationHeader encap;
          Encoding enc;
          if (!(ser >> encap) || !encap.to_encoding(enc, DCPS::MUTABLE) || enc.kind() != Encoding::KIND_XCDR1)) {
            if (DCPS::DCPS_debug_level > 0) {
              ACE_ERROR((LM_ERROR,
                        ACE_TEXT("(%P|%t) ERROR: Spdp::SpdpTransport::handle_input() - ")
                        ACE_TEXT("failed to deserialize encapsulation header for SPDP\n")));
            }
            return 0;
          }
          ser.encoding(enc);
```

I also tweaked the `encap.kind()` check because `to_encoding` does part of that for us, so we just need to check the base encoding kind.